llvm/include/llvm/CodeGen/ValueTypes.td

```diff
@@ -63,6 +63,7 @@ class VTVecTup<int size, int nf, ValueType dummy_elt, int value>
   let NF = nf;
   let ElementType = dummy_elt;
   let isRISCVVecTuple = true;
+  let isScalable = true;
```

Comment:
I don't think you should to this. This field is misnamed. It should be called "isScalableVector"

Reply:
Do you mean we should make tuple type a special case and handle it in switch in `getSizeInBits`?

Reply:
What I don't want is for this to return true for a tuples.

```
    bool isScalableVector() const {                                              
      return (SimpleTy >= MVT::FIRST_SCALABLE_VECTOR_VALUETYPE &&                
              SimpleTy <= MVT::LAST_SCALABLE_VECTOR_VALUETYPE);                  
    }  
```

I think you just need to change `Sc || Ty == aarch64svcount` in this to `Sc || Tup || Ty == aarch64svcount`

```
    TypeSize getSizeInBits() const {
      static constexpr TypeSize SizeTable[] = {
#define GET_VT_ATTR(Ty, N, Sz, Any, Int, FP, Vec, Sc, Tup, NF, NElem, EltTy) \
    TypeSize(Sz, Sc || Ty == aarch64svcount /* FIXME: Not in the td. */),
#include "llvm/CodeGen/GenVT.inc"                                                
#undef GET_VT_ATTR
```

Reply:
> What I don't want is for this to return true for a tuples.
> 
> ```
>     bool isScalableVector() const {                                              
>       return (SimpleTy >= MVT::FIRST_SCALABLE_VECTOR_VALUETYPE &&                
>               SimpleTy <= MVT::LAST_SCALABLE_VECTOR_VALUETYPE);                  
>     }  
> ```
> 
I see

> I think you just need to change `Sc || Ty == aarch64svcount` in this to `Sc || Tup || Ty == aarch64svcount`
> 
> ```
>     TypeSize getSizeInBits() const {
>       static constexpr TypeSize SizeTable[] = {
> #define GET_VT_ATTR(Ty, N, Sz, Any, Int, FP, Vec, Sc, Tup, NF, NElem, EltTy) \
>     TypeSize(Sz, Sc || Ty == aarch64svcount /* FIXME: Not in the td. */),
> #include "llvm/CodeGen/GenVT.inc"                                                
> #undef GET_VT_ATTR
> ```
Yeah, it's easier lol~